### PR TITLE
CMake build

### PR DESCRIPTION
**Pull request by PeterPetrik**
- removed qmake system
- added cmake for all platforms
- refactored dev docs - see INSTALL.md
- android APK is now multi-platform 
- qt 6.4.1
- qml compilation

### CODE REVIEW

**Comment by PeterPetrik** [comment]
@wonder-sk 
- You need new input-SDKs from today + qt 6.4.1
- look at docs/developer/platform for `cmake` commands for my env/pc
- macOS, ios: managed to build 
- linux: tried to fix Find*, not tried cmake/make yet
- windows: not started at all
- android: managed to run cmake generate, not tried building, CI not updated
- all platform: not started dist/package steps at all

**Comment by wonder-sk** [comment]
got the linux build working, updated linux dev docs. there are probably some issues with projections (as usual :laughing:)

File: scripts/format_qml.bash [hunk 1]
```diff
@@ -57,7 +57,7 @@ echo "Starting to format QML files"
 
 RETURN=0
 
-FILES=`find ../app ../qgsquick -name \*.qml* -print`
+FILES=`find ../app ../input -name \*.qml* -print`
```
Comment (tomasMizera) [inline scripts/format_qml.bash:60]:
```suggestion
FILES=`find ../app -name \*.qml* -print`
```

File: app/main.cpp [hunk 2]
```diff
@@ -312,33 +312,34 @@ void initDeclarative()
...
+  engine.addImportPath( ":/com.merginmaps/imports:" );
+
+  // This adds a runtime qml directory containing Input plugin
   // when Input is installed (e.g. Android/Win32)
   engine.addImportPath( QgsApplication::qmlImportPath() );
   qDebug() << "adding QML import Path: " << QgsApplication::qmlImportPath();
 
 #ifdef QML_BUILD_IMPORT_DIR
-  // Adds a runtime qml directory containing QgsQuick plugin
+  // Adds a runtime qml directory containing Input plugin
```
Comment (tomasMizera) [inline app/main.cpp:342]: Can we remove this `#ifdef QML_BUILD_IMPORT_DIR` and `#ifdef Q_OS_IOS` (below on line 350) altogether now?

Comment (PeterPetrik) [reply app/main.cpp:342]: ok

Comment (PeterPetrik) [reply app/main.cpp:342]: maybe you can fix it when merged to master?

**Pull request merged**